oeps/oep-0050-hooks-extension-framework.rst

```diff
@@ -0,0 +1,235 @@
+it, and return silently to the trigger location.
+Filters on the other hand are designed so that they are given some context data,
+act on it, and return a modified version for the rest of the core-code to use in
+the next steps of execution. We intend to make the two kinds of hooks
+interchangeable such that developers are not limited by the framework but
+instead given maximum flexibility on its usage to attain their extension goals.
+
+Hooks are used within WordPress to support almost all the forms in which it can
+be extended and are the basis of the very successful marketplace for plugins and
+themes. Given that Open edX already has a variety of theming capabilities and
+options we have opted here not to include any hooks that are meant to be used
+for theming.
+
+This proposal does not copy the model verbatim, since the open edX platform uses
+a different base technology which already allows for many extension points.
+This proposal deals primarily with the specific spots in the code execution that
+for many initiatives have been identified as the critical points that would
+allow them to adopt this technology without requiring modifications to the core.
+
+On the technical side, Django and the support for auto-installed plugins using
+stevedore already allow the addition of code in a very clean way. This proposal
+builds on top of this, only trying to standardize and grow the current
+capabilities of the platform.
+
+.. _two types of hooks: https://developer.wordpress.org/plugins/hooks/#actions-vs-filters
+
+
+Performance Considerations
+--------------------------
+
+This proposal is a double-edged sword. By allowing plugins to install code on so
+many and so varied places of the application performance could be completely
+destroyed to the point of making the platform unusable.
+
+This is a very important topic, but we believe that any responsible
+implementation of a hook must keep its own consequences and risks in check.
+In other words, it is up to the users of the hooks framework to be mindful of
+the performance of the platform by connecting to the appropriate triggers and
+optimizing their actions to the appropriate scale their Open edX instance runs on.
+
+Using asynchronous tasks to fulfill the actions and moreover connecting actions
+to the triggers in a way that is directly asynchronous is strongly recommended.
+The initial implementation must include sufficient tooling to make this possible
+and encourage it.
+
+Alternatively, in the core platform code, the actual exposition of each
+particular trigger could be configured (via flags or settings) so that even for
+scales at which the simple lookup for the actions that might be connected to a
+particular trigger could have performance implications is accounted for.
+
+
+Backward Compatibility
+======================
+
+The only extension point that will be affected by this OEP is the
+REGISTRATION_EXTENSION_FORM. This extension point should still be supported
```

Comment:
[clarification] Is the proposal to reimplement the current extensibility provided by `REGISTRATION_EXTENSION_FORM` to instead use this new framework?

Reply:
I made it clear now. Our suggestion is to keep the current feature until it is clear that it can be implemented using hooks and a migration path has been documented